workers/scanner.go

```diff
@@ -171,12 +171,30 @@ func (s *Scanner) pinUnderpinnedSkylinks() {
 		default:
 		}
 
-		skylink, sp, continueScanning := s.findAndPinOneUnderpinnedSkylink()
+		skylink, sp, continueScanning, err := s.findAndPinOneUnderpinnedSkylink()
 		if !continueScanning {
 			return
 		}
-		// Block until the pinned skylink becomes healthy or until a timeout.
-		s.waitUntilHealthy(skylink, sp)
+		// We only check the error if we want to continue scanning. The error is
+		// already logged and the only indication it gives us is whether we
+		// should wait for the file we pinned to become healthy or not. If there
+		// is an error, then there is nothing to wait for.
+		if err == nil {
+			// Block until the pinned skylink becomes healthy or until a timeout.
+			s.waitUntilHealthy(skylink, sp)
+		} else {
```

Comment:
nit: I would put a `continue` after the `s.waitUntilHealthy` and then remove the else to have the select de-indented.

Reply:
Might as well handle it right away. I used to opportunity to fix some typos and such as well.